### PR TITLE
Show welcome screens

### LINKED ISSUE #983: Suggestions view: Welcome screen for new users
See #513 

Atm it shows the empty state screen which is meant for returning users.

### LINKED ISSUE #984: Log view: Welcome screen for new users
See #514

Atm it shows the empty state screen which is meant for returning users.

### PR DESCRIPTION
**Pull request by willsb**
Closes #983 
Closes #984 

Squash whenever

### CODE REVIEW

File: Toggl.Foundation.Tests/Generators/ViewModelGenerators.cs [hunk 1]
```diff
@@ -25,10 +26,11 @@ public static Gen<TimeEntriesLogViewModel> ForTimeEntriesLogViewModel(MonthPredi
                 .Generator
                 .Select(dateTimes =>
                 {
-                    var source = Substitute.For<ITogglDataSource>();
+                   var source = Substitute.For<ITogglDataSource>();
                     var timeService = Substitute.For<ITimeService>();
+                    var onboardingStorage = Substitute.For<IOnboardingStorage>();
                     var navigationService = Substitute.For<IMvxNavigationService>();
-                    var viewModel = new TimeEntriesLogViewModel(source, timeService, navigationService);
+                var viewModel = new TimeEntriesLogViewModel(source, timeService, onboardingStorage, navigationService);
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/Generators/ViewModelGenerators.cs]: This formatting is weird.

File: Toggl.Daneel/Services/UserDefaultsStorage.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        public void SetUnauthorizedAccess(string apiToken)
+        {
+            setString(unauthorizedAccessKey, apiToken);
+        }
+
+        public bool IsClientOutdated()
+            => isOutdated(outdatedClientKey);
+
+        public bool IsApiOutdated()
+        => isOutdated(outdatedApiKey);
```
Comment (simonrozsival) [inline Toggl.Daneel/Services/UserDefaultsStorage.cs]: Indentation.

File: Toggl.Daneel/Services/UserDefaultsStorage.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+        public bool CompletedOnboarding() => getBool(completedOnboardingKey);
+
+        #endregion
+
+        private bool getBool(string key)
+            => NSUserDefaults.StandardUserDefaults.BoolForKey(key);
+
+        private string getString(string key)
+        => NSUserDefaults.StandardUserDefaults.StringForKey(key);
```
Comment (simonrozsival) [inline Toggl.Daneel/Services/UserDefaultsStorage.cs]: Indentation.

File: Toggl.Daneel/Services/UserDefaultsStorage.cs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+        private Version getStoredVersion(string key)
+        {
+            var stored = getString(key);
+            return stored == null ? null : Version.Parse(stored);
+        }
+
+        #endregion
+
+        #region IAccessRestrictionStorage
```
Comment (simonrozsival) [inline Toggl.Daneel/Services/UserDefaultsStorage.cs]: Wrong region name.

File: Toggl.Daneel/Services/UserDefaultsStorage.cs [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+            var now = timeService.CurrentDateTime;
+            var lastUsed = getString(lastAccessDateKey);
+            setString(lastAccessDateKey, now.ToString());
+
+            if (lastUsed == null) return;
+
+            var lastUsedDate = DateTimeOffset.Parse(lastUsed);
+            var offset = now - lastUsedDate;
+            if (offset < TimeSpan.FromDays(60)) return;
```
Comment (simonrozsival) [inline Toggl.Daneel/Services/UserDefaultsStorage.cs]: Magic constant 60 - please give the `TimeSpan` some name to document what it means.

File: Toggl.Foundation.MvvmCross/ViewModels/SuggestionsViewModel.cs [hunk 6]
```diff
@@ -30,20 +28,27 @@ public sealed class SuggestionsViewModel : MvxViewModel
         [DependsOn(nameof(Suggestions))]
         public bool IsEmpty => !Suggestions.Any();
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/SuggestionsViewModel.cs]: Do we even need it any more? From what I found the only usage is in this file on line 33.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntriesLogViewModelTests.cs [hunk 7]
```diff
@@ -49,17 +53,21 @@ public void ThrowsIfAnyOfTheArgumentsIsNull(bool useDataSource, bool useTimeServ
...
+                OnboardingStorage.IsNewUser().Returns(true);
+
+                await ViewModel.Initialize();
 
                 ViewModel.EmptyStateTitle.Should().Be(Resources.TimeEntriesLogEmptyStateWelcomeTitle);
             }
 
             [Fact, LogIfTooSlow]
-            public void ReturnsTheDefaultStringIfTheIsWelcomePropertyIsFalse()
+            public async ThreadingTask ReturnsTheDefaultStringIfTheIsWelcomePropertyIsFalse()
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntriesLogViewModelTests.cs]: Outdated test name.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntriesLogViewModelTests.cs [hunk 8]
```diff
@@ -68,17 +76,21 @@ public void ReturnsTheDefaultStringIfTheIsWelcomePropertyIsFalse()
         public sealed class TheEmptyStateTextProperty : TimeEntriesLogViewModelTest
         {
             [Fact, LogIfTooSlow]
-            public void ReturnsTheWelcomeStringIfTheIsWelcomePropertyIsTrue()
+            public async ThreadingTask ReturnsTheWelcomeStringIfTheIsWelcomePropertyIsTrue()
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntriesLogViewModelTests.cs]: Same here.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntriesLogViewModelTests.cs [hunk 9]
```diff
@@ -68,17 +76,21 @@ public void ReturnsTheDefaultStringIfTheIsWelcomePropertyIsFalse()
...
+                OnboardingStorage.IsNewUser().Returns(true);
+
+                await ViewModel.Initialize();
 
                 ViewModel.EmptyStateText.Should().Be(Resources.TimeEntriesLogEmptyStateWelcomeText);
             }
 
             [Fact, LogIfTooSlow]
-            public void ReturnsTheDefaultStringIfTheIsWelcomePropertyIsFalse()
+            public async ThreadingTask ReturnsTheDefaultStringIfTheIsWelcomePropertyIsFalse()
```
Comment (simonrozsival) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/TimeEntriesLogViewModelTests.cs]: Same here.

File: Toggl.Daneel/ViewSources/SuggestionsTableViewSource.cs [hunk 10]
```diff
@@ -71,7 +75,12 @@ public override UITableViewCell GetCell(UITableView tableView, NSIndexPath index
         public override nint NumberOfSections(UITableView tableView) => 1;
 
         public override nint RowsInSection(UITableView tableview, nint section)
-            => ItemsSource.Count().Clamp(0, maxNumberOfSuggestions);
+        {
+            var count = ItemsSource.Count();
+            if (IsWelcome && count == 0) return 3;
```
Comment (simonrozsival) [inline Toggl.Daneel/ViewSources/SuggestionsTableViewSource.cs]: Magical constant 3 - please give it a name.

File: Toggl.PrimeRadiant/Settings/IOnboardingStorage.cs [hunk 11]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Toggl.PrimeRadiant.Settings
+{
+    public interface IOnboardingStorage
+    {
+        void SetIsNewUser(bool isNewUser);
+        void SetCompletedOnboarding();
```
Comment (simonrozsival) [inline Toggl.PrimeRadiant/Settings/IOnboardingStorage.cs]: This is never used in this PR.

File: Toggl.PrimeRadiant/Settings/IOnboardingStorage.cs [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+
+        bool IsNewUser();
+        bool CompletedOnboarding();
```
Comment (simonrozsival) [inline Toggl.PrimeRadiant/Settings/IOnboardingStorage.cs]: This is also never used in this PR.

File: Toggl.PrimeRadiant/Settings/IOnboardingStorage.cs (hunk 11 above)
Comment (willsb) [reply Toggl.PrimeRadiant/Settings/IOnboardingStorage.cs]: I just put this there because we'll eventually need it. Can remove if you really want to

Comment (simonrozsival) [reply Toggl.PrimeRadiant/Settings/IOnboardingStorage.cs]: Don't remove it, I take this one back :)

**Comment by willsb** [comment]
The base branch has been changed because those fixes need to be included in the 1.0 release

**Pull request merged**